llvm/lib/Target/AMDGPU/AMDGPUInsertSingleUseVDST.cpp

```diff
@@ -83,26 +84,13 @@ class AMDGPUInsertSingleUseVDST : public MachineFunctionPass {
         // instruction to be marked as a single use producer.
         bool AllProducerOperandsAreSingleUse = true;
 
-        for (const auto &Operand : MI.operands()) {
-          if (!Operand.isReg())
-            continue;
-          const auto Reg = Operand.getReg();
-
-          // Count the number of times each register is read.
-          if (Operand.readsReg())
-            for (const MCRegUnit &Unit : TRI->regunits(Reg))
-              RegisterUseCount[Unit]++;
+        // Gather a list of Registers used before updating use counts to avoid
+        // double counting registers that appear multiple times in a single
+        // MachineInstr.
+        DenseSet<MCRegUnit> RegistersUsed;
```

Comment:
A DenseSet feels like a heavy way to track this

Reply:
I didn't realise DenseSet wasn't ideal, I chose it as the elements need to be unique. Is there a better alternative that you would suggest?